AdvancedExample/create.ps1

```diff
@@ -0,0 +1,355 @@
+        }
+    }
+
+    $actionMessage = "Creating session"
+
+    $teamsSessionSplatParams = @{
+        TenantId     = $actionContext.Configuration.TenantID
+        ClientId     = $actionContext.Configuration.AppId
+        ClientSecret = $actionContext.Configuration.AppSecret
+    }
+
+    $teamsSession = New-TeamsSession @teamsSessionSplatParams
+
+
+    $actionMessage = "verifying correlation configuration and properties"
+
+    if ($actionContext.CorrelationConfiguration.Enabled -eq $true) {
+        if ([string]::IsNullOrEmpty($correlationField)) {
+            throw "Correlation is enabled but not configured correctly."
+        }
+
+        if ([string]::IsNullOrEmpty($correlationValue)) {
+            throw "The correlation value for [$correlationField] is empty. This is likely a mapping issue."
+        }
+
+        if ([string]::IsNullOrEmpty($account.LocationCode)) {
+            Throw "The Location Code is empty for person"
+        }
+
+        #region Get Microsoft Teams Voip assignment
+        Write-Verbose "Querying MS Teams Phonenumber Assignment where [AssignedPstnTargetId] = [$($account.userPrincipalName)] and [NumberType] = [$($account.PhoneNumberType)]"
+
+        $getPhonenumberAssignmentSplatParams = @{
+            AssignedPstnTargetId = $account.userPrincipalName
+            NumberType           = $account.PhoneNumberType
+            Verbose              = $false
+            ErrorAction          = "Stop"
+        }
+
+        $currentPhonenumberAssignment = Get-CsPhoneNumberAssignment @getPhonenumberAssignmentSplatParams
+
+        Write-Verbose "Successfully queried MS Teams Phonenumber Assignment where [AssignedPstnTargetId] = [$($account.userPrincipalName)] and [NumberType] = [$($account.PhoneNumberType)]. Result count: $(($currentPhonenumberAssignment | Measure-Object).Count)" # Result: $($currentPhonenumberAssignment | ConvertTo-Json)"
+
+        if (($currentPhonenumberAssignment | Measure-Object).count -eq 0) {
+            # No phone number assignment found, get Teams user for correlation
+            $getTeamsUserSplatParams = @{
+                Identity          = $account.userPrincipalName
+                Verbose           = $false
+                ErrorAction       = "Stop"
+            }
+
+            $currentTeamsUser = Get-CsOnlineUser @getTeamsUserSplatParams
+            $returnObjectCount = $currentTeamsUser
+            $returnObject = [PSCustomObject]@{
+                Id = $currentTeamsUser.Identity
+                PhoneNumber = $null
+            }
+        } else {
+            $returnObjectCount = $currentPhonenumberAssignment
+            $returnObject = [PSCustomObject]@{
+                Id = $currentPhonenumberAssignment.AssignedPstnTargetId
+                PhoneNumber = $currentPhonenumberAssignment.TelephoneNumber
+            }
+        }
+    }
+    else {
+        if ($actionContext.Configuration.correlateOnly -eq $true) {
```

Comment:
Zo te zien gebruiken we deze optie verder nergens dus zou weg kunnen?

Reply:
yup, copy/paste foutje. Wel in origineel, niet nodig in deze versie